torchaudio/__init__.py

```diff
@@ -1,33 +1,61 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
+
 import os.path
+import warnings
 
 import torch
-import _torch_sox
 
-from torchaudio import transforms, datasets, kaldi_io, sox_effects, compliance
+from torchaudio import (
+    _soundfile_backend,
+    compliance,
+    datasets,
+    kaldi_io,
+    sox_backend,
+    sox_effects,
+    transforms,
+)
 
 try:
     from .version import __version__, git_version  # noqa: F401
 except ImportError:
     pass
 
 
-def check_input(src):
-    if not torch.is_tensor(src):
-        raise TypeError('Expected a tensor, got %s' % type(src))
-    if src.is_cuda:
-        raise TypeError('Expected a CPU based tensor, got %s' % type(src))
+_audio_backend = "sox"
+
+
+def set_audio_backend(backend):
+    """
+    Specifies the package used to load.
+    Args:
+        backend (string): Name of the backend. one of {'sox'}.
```

Comment:
nit: add soundfile

Having references to the sources of those backends in the doctstring could be useful too